Draft adjustments for the README #517

TobiasNx opened the pull request:
@fsteeg and @dr0i could you have a look. Maybe @dr0i first.

TobiasNx committed:
Draft adjustments for the README #517

TobiasNx committed:
Add stage step and some TODOs #517

dr0i commented:
Check the following to understand if the `sbt stage` is needed:
Find your `fix`. If it's also in the `target` directory you can see by diffing the source `fix` and the "`target`-`fix`" that they differ. If that's the case, then yes, you will have to rebuild the app.

TobiasNx committed:
Merge remote-tracking branch 'origin/master' into 517-readmeSbt

TobiasNx committed:
Simplify the steps for local web app build #517

TobiasNx commented:
Yes all changes in the fix, need to be staged, should we document this! Also could you review and finalize this?

dr0i commented on README.md:
```diff
@@ -53,12 +44,20 @@ To get the lookup table `conf/wikidataLookup.tsv`:
 - `cd app/transformation/input/`
-- `wget http://quaoar1.hbz-nrw.de:7001/assets/data/dbs.zip; unzip dbs.zip`
-- `wget http://quaoar1.hbz-nrw.de:7001/assets/data/sigel.xml`
+- `wget http://quaoar1.hbz-nrw.de:7001/assets/data/dbs.zip; unzip dbs.zip` TODO: this is not correct anymore!
```
What _is_ the correct one, then?

TobiasNx replied:
Yes `app/transformation/input/dbs.csv` and `app/transformation/input/sigel.dat` are the files needed. The zip file is not used anymore as well as the xml file for the isil data is not used anymore instead we are using the PICA data.

How should I change the instructions here then?

dr0i commented on README.md:
```diff
@@ -132,7 +115,15 @@ lobid-organisations is a web app implemented with Play to serve the JSON-LD cont
+
+- `sbt clean`
+- `sbt --java-home $JAVA_HOME stage`
+- `JAVA_OPTS="$JAVA_OPTS -XX:+ExitOnOutOfMemoryError" ./target/universal/stage/bin/lobid-organisations -Dhttp.port=9000 -no-version-check`
```
These steps are mentioned before. Only difference is the port number. You could reference the already existing recipe and explain why to change the port.

TobiasNx replied:
One does not need the `--java-home` reference for staging or the other additional parameters for startin the web app locally.

TobiasNx committed:
Get rid of all morph related stuff #517

This will help with the documentation since the any info to morph is not needed anymore.

TobiasNx commented:
@dr0i unfortunately the transformation does not run anymore after the deletion of the morph related stuff:

```
$ ./target/universal/stage/bin/lobid-organisations -no-version-check
OpenJDK 64-Bit Server VM warning: Ignoring option MaxPermSize; support was removed in 8.0
Play server process ID is 136097
SLF4J: Class path contains multiple SLF4J bindings.
SLF4J: Found binding in [jar:file:/home/tobias/git/lobid-organisations/target/universal/stage/lib/ch.qos.logback.logback-classic-1.1.1.jar!/org/slf4j/impl/StaticLoggerBinder.class]
SLF4J: Found binding in [jar:file:/home/tobias/git/lobid-organisations/target/universal/stage/lib/org.slf4j.slf4j-simple-1.7.36.jar!/org/slf4j/impl/StaticLoggerBinder.class]
SLF4J: Found binding in [jar:file:/home/tobias/git/lobid-organisations/target/universal/stage/lib/org.slf4j.slf4j-reload4j-1.7.36.jar!/org/slf4j/impl/StaticLoggerBinder.class]
SLF4J: See http://www.slf4j.org/codes.html#multiple_bindings for an explanation.
SLF4J: Actual binding is of type [ch.qos.logback.classic.util.ContextSelectorStaticBinder]
[info] application - Starting transformation, will write to '/tmp/lobid-organisations/enriched.prod.out.json' 
resumptionToken: 
resumptionToken: fBSWeNo9i0EKgDAQA19i8AOF3dbdtB_w7Be0KAh6EsHnqwfNLZOJ0uklirhENsNax2E7j3DtG6Z1Qg8TKF2L2SM9QYvmXfQ7lhf-jcxMkRmdmM9LzUGZRyiS2w2Dtxja
resumptionToken: fsJXeNo1i0sKgDAQQ09i8ALCTGsn7QVc9wqtKAi6EsHj-0Gzy8uL0mjJiZg4NnkZS16PvTu3FXWpGKAiUJqmEG7rDlo0z6TfU1_4NyEjvWNEL8GmeYydMhYovIULmKwY_g
resumptionToken: fpCyeNo1i8sJgEAQQysx2IAwszqTtQHPtrCKwsJ6EsHy_aC55eVF6fQ-iLgEVmOe01iOvTm3gilPGKAmULr2Zrd1BzWqZ9LvqS_8m5CRbWBEJ-bLOsdGGRMUrdsFmiMZAw
resumptionToken: fLFueNo1i0EKgDAQA19i8AOF3dZu6gc89wtaFAQ9ieDzraI5JTNEabTei5h4NnktY97Ow137hmmdMKAqKE37GGutQYvmUfo99YX_EjIxeCZ0Em1eSnLKNEIRLN6Y-Bj_
resumptionToken: fhYseNo1i8sJgEAQQysx2IAwszqTtQHPtrCKwsJ6EsHy_aC55eVF6fQ-iLgEVmOe01iOvTm3gilPGBBMoHTtzW7rDmpUz6TfU1_4NyEj28CITsyXdY6NMiYoWrcLmm8ZBA
resumptionToken: flfXeNotjMEJgDAQBCtxsYHAXWJukwZ8p4UoCgF9iWD5KjrfnR2l0bIXMfHsSptr2c7DXfuGqU0YEUSgNM0xPtYDenTvpP9TVT7l65CJwTNhkGjLOienTBWKYPEGmW4ZAQ
resumptionToken: flWteNo1i0EKgDAQA19i8APCbutu6gc89wu2KAh6EsHnq6C5ZTJROn0IIi6BTV7rlLfz6K59Q1kLRkQTKF0Hs8d6ghbNO-n3VH3h34RMjIEJvZjPS02dMk1QRLcbmuUZBg
resumptionToken: 
[error] ROOT - Transformation failed
org.metafacture.framework.MetafactureException: java.io.IOException: Remotely Closed
        at org.metafacture.io.LineReader.process(LineReader.java:74) ~[org.metafacture.metafacture-io-7.0.0.jar:na]
        at org.metafacture.io.LineReader.process(LineReader.java:55) ~[org.metafacture.metafacture-io-7.0.0.jar:na]
        at org.metafacture.io.LineReader.process(LineReader.java:37) ~[org.metafacture.metafacture-io-7.0.0.jar:na]
        at org.metafacture.io.FileOpener.process(FileOpener.java:158) ~[org.metafacture.metafacture-io-7.0.0.jar:na]
        at transformation.TransformDbs.process(TransformDbs.java:41) ~[lobid-organisations.lobid-organisations-1.0-SNAPSHOT.jar:1.0-SNAPSHOT]
        at transformation.TransformAll.process(TransformAll.java:59) ~[lobid-organisations.lobid-organisations-1.0-SNAPSHOT.jar:1.0-SNAPSHOT]
        at controllers.Transformation.transformSet(Transformation.java:46) ~[lobid-organisations.lobid-organisations-1.0-SNAPSHOT.jar:1.0-SNAPSHOT]
        at Global.initData(Global.java:55) ~[lobid-organisations.lobid-organisations-1.0-SNAPSHOT.jar:na]
        at Global.onStart(Global.java:31) ~[lobid-organisations.lobid-organisations-1.0-SNAPSHOT.jar:na]
        at play.core.j.JavaGlobalSettingsAdapter.onStart(JavaGlobalSettingsAdapter.scala:22) ~[com.typesafe.play.play_2.11-2.3.9.jar:2.3.9]
Caused by: java.io.IOException: Remotely Closed
0    [main] INFO               org.elasticsearch.node  - [Graydon Creed] version[2.3.2], pid[136097], build[b9e4a6a/2016-04-21T16:03:47Z]
1    [main] INFO               org.elasticsearch.node  - [Graydon Creed] initializing ...
12   [main] INFO            org.elasticsearch.plugins  - [Graydon Creed] modules [], plugins [zero, location-aggregation, bundle], sites []
WARNING: An illegal reflective access operation has occurred
WARNING: Illegal reflective access by org.apache.lucene.store.MMapDirectory$1 (file:/home/tobias/git/lobid-organisations/target/universal/stage/lib/org.apache.lucene.lucene-core-5.5.0.jar) to method java.nio.DirectByteBuffer.cleaner()
WARNING: Please consider reporting this to the maintainers of org.apache.lucene.store.MMapDirectory$1
WARNING: Use --illegal-access=warn to enable warnings of further illegal reflective access operations
WARNING: All illegal access operations will be denied in a future release
43   [main] INFO                org.elasticsearch.env  - [Graydon Creed] using [1] data paths, mounts [[/ (/dev/nvme0n1p5)]], net usable_space [103.9gb], net total_space [375.8gb], spins? [no], types [ext4]
43   [main] INFO                org.elasticsearch.env  - [Graydon Creed] heap size [1gb], compressed ordinary object pointers [true]
1766 [main] INFO               org.elasticsearch.node  - [Graydon Creed] initialized
1766 [main] INFO               org.elasticsearch.node  - [Graydon Creed] starting ...
1770 [main] INFO          org.elasticsearch.transport  - [Graydon Creed] publish_address {local[1]}, bound_addresses {local[1]}
1773 [main] INFO          org.elasticsearch.discovery  - [Graydon Creed] elasticsearch/gyqeBCNdSviZ-amZycXtNw
1780 [elasticsearch[Graydon Creed][clusterService#updateTask][T#1]] INFO    org.elasticsearch.cluster.service  - [Graydon Creed] new_master {Graydon Creed}{gyqeBCNdSviZ-amZycXtNw}{local}{local[1]}{local=true}, reason: local-disco-initial_connect(master)
1833 [main] INFO               org.elasticsearch.http  - [Graydon Creed] publish_address {127.0.0.1:7210}, bound_addresses {127.0.0.1:7210}
1833 [main] INFO               org.elasticsearch.node  - [Graydon Creed] started
1846 [elasticsearch[Graydon Creed][clusterService#updateTask][T#1]] INFO            org.elasticsearch.gateway  - [Graydon Creed] recovered [0] indices into cluster_state
Oops, cannot start the server.
java.lang.IllegalArgumentException: File /tmp/lobid-organisations/enriched.prod.out.json is not large enough - should be >='15000000' but is 47
        at controllers.Index.initialize(Index.java:124)
        at Global.initData(Global.java:57)
        at Global.onStart(Global.java:31)
        at play.core.j.JavaGlobalSettingsAdapter.onStart(JavaGlobalSettingsAdapter.scala:22)
        at play.api.GlobalPlugin.onStart(GlobalSettings.scala:220)
        at play.api.Play$$anonfun$start$1$$anonfun$apply$mcV$sp$1.apply(Play.scala:91)
        at play.api.Play$$anonfun$start$1$$anonfun$apply$mcV$sp$1.apply(Play.scala:91)
        at scala.collection.immutable.List.foreach(List.scala:392)
        at play.api.Play$$anonfun$start$1.apply$mcV$sp(Play.scala:91)
        at play.api.Play$$anonfun$start$1.apply(Play.scala:91)
        at play.api.Play$$anonfun$start$1.apply(Play.scala:91)
        at play.utils.Threads$.withContextClassLoader(Threads.scala:21)
        at play.api.Play$.start(Play.scala:90)
        at play.core.StaticApplication.<init>(ApplicationProvider.scala:55)
        at play.core.server.NettyServer$.createServer(NettyServer.scala:253)
        at play.core.server.NettyServer$$anonfun$main$3.apply(NettyServer.scala:289)
        at play.core.server.NettyServer$$anonfun$main$3.apply(NettyServer.scala:284)
        at scala.Option.map(Option.scala:146)
        at play.core.server.NettyServer$.main(NettyServer.scala:284)
        at play.core.server.NettyServer.main(NettyServer.scala)
```

In contrast the current master runs smoothly without errors.

PS: I merged the master and it works now. So this is outdated.

TobiasNx committed:
Merge remote-tracking branch 'origin' into 517-readmeSbt

dr0i committed:
Add where the data comes from and how to get it (#528)

- remove redundant section of how to start/deploy the app

dr0i committed:
Restructure (#528)

dr0i commented on README.md:
```diff
@@ -53,12 +44,20 @@ To get the lookup table `conf/wikidataLookup.tsv`:
-- `wget http://quaoar1.hbz-nrw.de:7001/assets/data/dbs.zip; unzip dbs.zip`
-- `wget http://quaoar1.hbz-nrw.de:7001/assets/data/sigel.xml`
+- `wget http://quaoar1.hbz-nrw.de:7001/assets/data/dbs.zip; unzip dbs.zip` TODO: this is not correct anymore!
+- `wget http://quaoar1.hbz-nrw.de:7001/assets/data/sigel.xml` TODO: this is not correct anymore!
```
As these are not open data I've just removed the section. I've added how we get the data, though.

dr0i reviewed (approved):
I've changed:
- how we get the source data
- remove redundant sections
- restructure a bit
